### PR TITLE
Nuke Ops updates and buffs

### PR DESCRIPTION
**Pull request by Markolie**
**What does this PR do:**
This pull request makes a number of changes to the Nuke Ops gamemode (most of which has been ported from /tg/).

1. The equipment lockers on the Syndicate ship have been replaced with suit storage units. Suit storage units have been given a separate magboots slot to accommodate this.
2. Additional spawn nuclear operative spawn points have been added to their base, in case we ever increase their spawn number.
3. A new category has been added: "Support and Mechanized Exosuits". The exosuits have been moved to this category. 
4. Instead of buying a singular cyborg spawner, it has been separated into three purchases. The Assault Cyborg is now 65 telecrystals (from 50), the Medical Cyborg is now 35 telecrystals and the Saboteur Cyborg is now 25 telecrystals. Saboteurs now spawn with thermals.
5. Operatives can now purchase reinforcements. One reinforcement costs 25 telecrystals and comes with basic equipment.
6. A new category has been added "Bundles and Telecrystals". Existing bundles and telecrystals have been moved to this category. Ammo bundles are still under the "Ammunition" category.
7. Spawning Syndicate operatives through the traitor panel now properly updates their ID.
8. The telecrystal scaling modifier has been set to 1.2. This means that for every ten players on the server the operatives now receive 2 telecrystals more overall (eventually rounded up). So at 100 players, the operatives now have 20 extra telecrystals to spend among all of them.
9. The following bundles have been added/modified:

**Bulldog Bundle**
A bulldog, two 12g buckshot drums and thermal goggles for 9 telecrystals (normally 12).

**C-20r Bundle**
A C-20r, two magazines and a suppressor for 14 telecrystals (normally 17).

**Medical Bundle**
A tactical medkit, Donksoft LMG and box of riot darts for 15 telecrystals (normally 20).

**Sniper Bundle**
A sniper, two knockout magazines and a dark black suit with red tie for 18 telecrystals (normally 23).

**Bulldog Ammo Bundle** (modified)
Six 12g slugs, two 12g buckshots and one 12g dragon breath for 18 telecrystals (normally 12).

**C-20r Ammo Bundle**
Ten .45 magazines for 14 telecrystals (normally 20).

**C4 Bag**
Ten C4 charges for 8 telecrystals (normally 10).

**X4 Bag**
Three X4 charges for 4 telecrystals (normally 6).

**Changelog:**
:cl: Markolie
add: The hardsuit equipment lockers on the Syndicate Nuclear Operative shuttle have been replaced with suit storage units. Suit storage units now have a separate slot for magboots.
add: Nuclear Operatives can now purchase reinforcements. These reinforcements cost 25 telecrystals each.
add: A number of new ammunition and weapon bundles have been added for Nuclear Operatives.
tweak: For every ten players on the server, the Nuclear Operatives now receive 2 extra telecrystals.
tweak: The Syndicate borg spawners have been separated into three separate spawners, one for each type of Syndicate cyborg. Their prices have been modified.
tweak: Saboteur borgs now spawn with thermals.
tweak: The Syndicate borg and exosuits have been moved to a new category, "Support and Mechanized Exosuits".
tweak: Existing bundles and telecrystal purchases have been moved to a new category, "Bundles and Telecrystals".
tweak: There are now more spawn points for nuclear operatives on their base. The number of operatives spawning has not been changed.
fix: Spawning nuclear operatives through the traitor panel now causes their ID to be properly updated.
/:cl:

### CODE REVIEW

**Comment by MrMagolor** [comment]
You know the framework for extra nukeops is much more easily implemented now, given that there are no TC sharing consoles. Though there is still the matter of hardsuits and pinpointers...

**Comment by Markolie** [comment]
Yea, there's six armory lockers in the shuttle thankfully, so that's enough for everyone. And this was a very straight forward change to make, indeed.

**Comment by Fox-McCloud** [comment]
Eventually we should probably implement the "recruit" feature that TG has, where you can buy packs of weapons, get more recruits for dirt cheap, and what have you.

Nukies is just insufferable if the population is above 70.

**Comment by Darkmight9** [comment]
I'm afraid it's not so clear cut. The sixth locker on the syndi shuttle is the locker that contains the bulldog shotguns (and there's only 5 of those too in that locker). I can easily see the intelicard being moved over to the same table as the c4 though and add the extra locker there. 
image

**Comment by Markolie** [comment]
Ah, forgot about that. I’ll add another locker. I’ll also have a look at the /tg/ stuff tomorrow.

**Comment by Spartan6** [comment]
I’d rather have the recruit feature to this, or at least as well.

**Comment by Denthamos** [comment]
@Markolie Might want to look into replacing the lockers entirely with the suit storage units to hold the hardsuit/helmet, mask, and jetpack and then make each nukie that spawns in, automatically spawn with the pinpointer on them.  Less to figure out with grabbing things so nukies can gear up and go.

Also, its interesting how there is only 5 lockers for the hardsuits, yet 6 lockers for the general gear already in game.  IE the pistol mags, goggles, and belt.  5 below the nukie spawn area and 1 on the ship cockpit.

**Comment by Markolie** [comment]
Okay, the PR has been reworked to add the new ammo/weapon bundles and to add reinforcements. The additional Syndicate operative spawn has been removed; instead the number of telecrystals received per ten players has been increased.

Two things:
1. The assault cyborg has been increased in price to 65 (from 50) and the other borgs are cheaper now. Should we keep the assborg at 50?
2. Right now reinforcements spawn with absolutely no equipment and cost 25. We could increase the cost to 45 and let them spawn with regular nuclear operative gear (which includes 20 telecrystals). Thoughts?

Let me know if you'd like to see anything else changed in these bundles.

**Comment by TDSSS** [comment]
Absolutely no gear means no belt, no pinpointer, etc? Do they spawn totally naked?

**Comment by Markolie** [comment]
Nothing at all, correct.

**Comment by TDSSS** [comment]
Are there...spare jumpsuits at the nukie base? They probably need one to wear IDs and stuff. Also we disabled buying some stuff that nukies start with iirc so they don't waste their TC, might want to re-enable those so they can fully outfit the new guys

**Comment by Markolie** [comment]
The lockers in the base spawn with boots, belts and turtlenecks, so they can pick up the bare necessities there. Beyond that they will need to purchase everything (i.e. an agent ID).

**Comment by Denthamos** [comment]
Honestly if you are going to add in the reinforcement option, just do it the same way TG has it.  They get the basic spawn in gear ops get but no uplink.  Anything past that the roundstart nukies have to give them.

**Comment by TDSSS** [comment]
We have tried to reduce the potential for fuckups/flukeops in the nukies gamemode. Trying to outfit nukies that spawn completely without gear will really increase the potential for fuckups and turn equipping and coordinating into a mess.

You need to get them headsets with syndie keys so they can communicate, agent IDs so they can get through doors, masks and jetpacks, hardsuits, something to shoot...

I agree it would be much better to spawn them with basic gear. I'd like it if we even gave them an uplink with 0TC so that the other nukies can just give them some TC to equip themselves, instead of buying stuff for them.

**Comment by Markolie** [comment]
Sure, that's a super easy change.

**Comment by AzuleUtama** [comment]
With the bag of c4 explosives, both traitors and nukies can buy a pack of 5 for 4TC which equates to the same value as the bag of 10 for 8TC.

I'd either keep the pack as-is or replace it with the bag of 10 but allow traitors to also buy it.

**Comment by Markolie** [comment]
I removed the pack initially, but did decide to re-add it. I think it's less likely that traitors need 10 C4 and it's easier for traitors to use a box instead of a duffelbag as they can squash it, leaving now evidence. Nuke ops now have the choice of buying even more when they need it.

File: code/modules/antagonists/_common/antag_spawner.dm [hunk 1]
```diff
@@ -84,11 +127,29 @@
...
+	var/brainopslastname = pick(GLOB.last_names)
+	if(syndicate_name())  //the brain inside the syndiborg has the same last name as the other ops.
+		brainopslastname = syndicate_name()
+	var/brainopsname = "[brainfirstname] [brainopslastname]"
+
+	R.mmi.name = "[initial(R.mmi.name)]: [brainopsname]"
+	R.mmi.brainmob.real_name = brainopsname
+	R.mmi.brainmob.name = brainopsname
+
+	if (!switch_roles)
```
Comment (variableundefined) [inline code/modules/antagonists/_common/antag_spawner.dm]: Just spacingg

**Pull request merged**